nmoscommon/auth/nmos_auth.py

```diff
@@ -0,0 +1,173 @@
+import os
+import requests
+import json
+from requests.exceptions import RequestException
+from functools import wraps
+from flask import request
+try:
+    from urlparse import parse_qs
+except ImportError:
+    from urllib.parse import parse_qs
+from OpenSSL import crypto
+
+from nmoscommon.mdnsbridge import IppmDNSBridge
+from nmoscommon.nmoscommonconfig import config as _config
+from nmoscommon.logger import Logger as defaultLogger
+from authlib.specs.rfc7519 import jwt
+from authlib.specs.rfc6749.errors import MissingAuthorizationError, \
+    UnsupportedTokenTypeError
+from authlib.common.errors import AuthlibBaseError
+
+from .claims_options import IS_XX_CLAIMS
+from .claims_validator import JWTClaimsValidator
+from .constants import CERT_ENDPOINT, CERT_PATH, CERT_KEY
+
+MDNS_SERVICE_TYPE = "nmos-auth"
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+OAUTH_MODE = _config.get('oauth_mode', True)
+
+
+class RequiresAuth(object):
+
+    def __init__(self, condition=OAUTH_MODE, claimsOptions=IS_XX_CLAIMS,
+                 certificate=None):
+        self.condition = condition
+        self.claimsOptions = claimsOptions
+        self.certificate = certificate
+        self.bridge = IppmDNSBridge()
+        self.logger = defaultLogger("nmossecurity")
+
+    def getHrefFromService(self, serviceType):
```

Comment:
Doesn't necessarily need resolving here, but you have a mix of snake case and camel case. Python tends to prefer the former: https://www.python.org/dev/peps/pep-0008/#prescriptive-naming-conventions